test/parallel/test-stream2-finish-pipe.js

```diff
@@ -35,7 +35,11 @@ w._write = function(data, encoding, cb) {
 
 r.pipe(w);
 
-// This might sound unrealistic, but it happens in net.js. When
-// `socket.allowHalfOpen === false`, EOF will cause `.destroySoon()` call which
-// ends the writable side of net.Socket.
-w.end();
```

Comment:
I'm not sure if moving this to next tick invalidate the test. It's not clear to me what is the aim of the test.
Is it testing a `write()` by `readable.pipe()` before or after the writable stream is ended?

Reply:
I'm not sure I understand you question. Without nextTick this test should fail since you can't call `write` after `end`.

Reply:
Nvm, I've edited the comment and marked it as resolved.